subprojects/core/src/integTest/groovy/org/gradle/api/tasks/bundling/ArchiveIntegrationTest.groovy

```diff
@@ -307,13 +307,13 @@ class ArchiveIntegrationTest extends AbstractIntegrationSpec {
             task myTar(type: Tar) {
                 destinationDirectory = buildDir
 
-                assert compression == Compression.NONE
+                assert compression.get() == Compression.NONE
 
-                compression = Compression.GZIP
-                assert compression == Compression.GZIP
+                compression.set(Compression.GZIP)
```

Comment:
🤔 It should also work if we just keep it `compression = Compression.GZIP`, no?

Also for other setters